app/assets/javascripts/metrics.js

```diff
@@ -1,56 +1,226 @@
 // Loads all Semantic javascripts
 //= require semantic-ui
+
+// metrics api endpoints
+const metrics_endpoints = {
+	update: 'update_current_metrics',
+	get: 'get_current_metrics'
+}
+
+// prevents enumerator from being changed
+Object.freeze(metrics_endpoints);
+
 $(document).ready(function(){
-    $('.tabular.menu .item').tab();
-		$('.ui.dropdown').dropdown();
-		$('.ui.checkbox').checkbox();
-		$('.ui.calendar').calendar({type: 'date'});
-    $('.ui.form')
-	  .form({
-			inline: true,
-	    fields: {
-	      percentage1: {
-	        identifier: 'percentage1',
-	        optional: 'true',
-	        rules: [
-	          {
-	            type   : 'integer[0..100]',
-	            prompt : 'Please enter an integer from 0 to 100 for percentages'
-	          }
-	        ]
-				},
-				percentage2: {
-	        identifier: 'percentage2',
-	        optional: 'true',
-	        rules: [
-	          {
-	            type   : 'integer[0..100]',
-	            prompt : 'Please enter an integer from 0 to 100 for percentages'
-	          }
-	        ]
-				},
-	    }
-		});
-
-		$('[name="metrics-checkbox-1"]').change(function() {
-			$('#save').removeClass('disabled');
-		});
-
-		$('[name="metrics-checkbox-2"]').change(function() {
-			$('#save').removeClass('disabled');
-		});
-
-		$('[name="metrics-checkbox-3"]').change(function() {
-			$('#save').removeClass('disabled');
-		});
-
-		$('[name="metrics-checkbox-4"]').change(function() {
-			$('#save').removeClass('disabled');
-		});
-		
-		$(window).bind('beforeunload', function(){
-			if (!$('#save').hasClass('disabled')) {
-				return 'Make sure to save your changes';
+
+	// Initializing Fomantic UI elements
+    $('.tabular.menu .item').tab();	
+	$('.ui.dropdown').dropdown();
+	$('.ui.checkbox').checkbox();
+	$('.ui.calendar').calendar({type: 'date', initialDate: new Date()});
+	$('.ui.dropdown').dropdown('set selected',1);
+	$('#grade_drop_consecutive_counts').dropdown('set selected',2);
+	
+});
+
+// Updates form validation based on checked
+$('.checkbox').change(function(){
+
+	let fields = {};
+
+	if($('#grade_drop_checkbox').checkbox('is checked')){
+		fields['grade_drop_percentage'] = {
+			identifier: 'grade_drop_percentage',
+			rules: [
+			  {
+				type   : 'integer[1..100]',
+				prompt : 'Please enter an integer from 1 to 100 for percentages'
+			  }
+			]
+		};
+	}
+
+	if($('#low_grades_checkbox').checkbox('is checked')){
+		fields['low_grades_percentage'] = {
+			identifier: 'low_grades_percentage',
+			rules: [
+			  {
+				type   : 'integer[1..100]',
+				prompt : 'Please enter an integer from 1 to 100 for percentages'
+			  }
+			]
+		};
+	}
+	
+	$('.ui.form').form({inline: true, fields});
+
+});
+
+
+$.getJSON(metrics_endpoints['get'],function(data, status){
+	
+	if(status=='success'){
+
+		// situation when no conditions have been selected
+		if(data.length == 1 && data[0]['condition_type'] == "no_condition_selected")
```

Comment:
Question about this line: when I have no conditions selected I am receiving data.length == 0. Is this supposed to be when instructors have not set up any conditions for their class?